### DISCUSSION TITLE
Some ideas for future releases

### DISCUSSION THREAD

**Discussion by ghost** [diasurgical/DevilutionX#5837] [Ideas]
Hello, i really like Devx and have a few ideas that i think might be great for future releases.

**Single player quests and uniques in MP**
This feels like it was cut due to multiplayer being a last-minute thing. The uniques would deviate a bit from vanilla D1, but the quest uniques really aren't that great. Probably shouldn't drop every time you complete the quest, maybe just first time or only 10% of the time.

**Diablo death cinematic - only play first time**
It gets slightly annoying to see it every time you beat Diablo, it's gotten to the point where i just skip him to avoid the cinematic.

**Spellbook - list current spell level on them?**
It would be great if the current spell level of a spell was shown on the UI when i look at a new book.

**Elixir - Tell me if i'm capped on them.**
Same idea as the previous - Tell me the stat value. Don't want to waste my precious elixirs ;)

**Shrine information on them ala Diablo+**
Would be great if it just said what the shrine did, ala Diablo+. Tabbing out to check what it does interrupts the game.

**Spells - Tell me damage type on them.**
Would be great if the spells told me their damage type, it isn't always obvious.

**Show me spell hotkeys in the UI.**
I imagine an image of the spells and their hotkey somewhere on the UI.

**Spell hotkey mouse wheel.**
I sometimes catch myself trying to scroll to the next spell in my hotkey setup. Would be awesome if it actually worked.

**Floating damage number option**
Could be a cool option to have. I've seen footage of it, but i can't see a way to turn it on.

**Enemy type under health bar - Demon, undead, animal**
I'm loving the health bar setup. But i wish it also had information on the enemy type as well.

**Remove grief - Black deaths, Bad shrines, broken items.**
There are some incredibly griefy things in diablo 1 that will just ruin your day and/or character. The game would just be much better without them.

**Allow f1-f4 for spell hotkeys.**
Even after all these years, having the hotkeys be F4-F8 just seems so counterintuitive.

**Level up message when my stats are capped.**
The experience bar is really cool, but sometimes i just forget that i gained a level after my stats are capped. Some kind of short symbol or a sound when you level up would be awesome.

**Undropable unique fix?**
There are a few uniques in Diablo 1 that are nearly impossible to get because some other unique needs to spawn first. Would be great if this was fixed.

**Auto-sort inventory**
This is a feature from Titan Quest. It is just a button in the inventory that sorts everything, freeing up space effectively.

**Unique bow - Flame dart** 
Fire damage listed twice in description. It is effectively just a graphical bug.

Oh and i also had one question:
What does the little square that turns red on the monsters mean? Is it for damaged monsters? I can't quite figure it out.

**Comment by DakkJaniels** [comment]
I believe most of these suggestions have already been implemented or are in progress.

In 1.4.1, you should already be able to turn off being able to click bad shrines, and you should be able to assign up to 12 quick spell hotkeys.

Red square in the health bar at top of screen indicates who has hit the monster so far, for the purposes of splitting experience on the kill.

**Reply by ghost** [reply]
> Red square in the health bar at top of screen indicates who has hit the monster so far, for the purposes of splitting experience on the kill.

May i suggest changing it to a drop of blood instead?

It could also be cool if Cain could tell me what monster/object dropped my items.

Looking forward to the next version, you guys are doing great!

**Reply by DakkJaniels** [reply]
> May i suggest changing it to a drop of blood instead?

Feel free to create the drop of blood asset (or locate one that would be useable from a license perspective) and submit it as a pull request. I don't see how that would be any better, necessarily, at telling you what it means though.

**Reply by ghost** [reply]
Sure I can do that. Three questions:

1. Is standard RGB fine or do i need another colour palette?
2. What image file is best?
3. What dimensions do you guys want?

Blood is commonly associated with damage - Hence why i think a drop of blood would be a better indicator of the function.

**Reply by AJenbo** [reply]
1 You need to use the 128 constant colors available in the D1 palette.
2 PCX is probably the best middle ground as things stand right now.
3 Your the artist, figure out what size and placement it should have to look good

**Reply by ghost** [reply]
What size is the current one?

**Reply by AJenbo** [reply]
https://github.com/diasurgical/devilutionx-assets/blob/main/bundled-assets/data/monstertags.pcx

**Comment by AJenbo** [comment]
> Single player quests and uniques in MP

1.5.0
> Diablo death cinematic - only play first time

no
> Spellbook - list current spell level on them?

maybe
> Elixir - Tell me if i'm capped on them.

probably not
> Shrine information on them ala Diablo+

maybe
> Spells - Tell me damage type on them.

maybe
> Show me spell hotkeys in the UI.

open speed book
> Spell hotkey mouse wheel.

planned
> Floating damage number option

1.5.0
> Enemy type under health bar - Demon, undead, animal

enable in settings
> Remove grief - Black deaths, Bad shrines, broken items.

no, enable in settings, no
> Allow f1-f4 for spell hotkeys.

enable in settings
> Level up message when my stats are capped.

not so sure
> Undropable unique fix?

Not easy/unlikely
> Auto-sort inventory

maybe
> Unique bow - Flame dart

Submit an issues with a screenshot and a save game

**Reply by ghost** [reply]
>> Single player quests and uniques in MP

>1.5.0

Great! I have some ideas for how to implement the drop chance. I can make a separate post with them. Feel free to say yes or no.

>> Diablo death cinematic - only play first time

>no

Is it because hard to do or because bad idea?

I have two ideas on how to skip the cinematic. Both involve storing a binary (1 for "Diablo slain", 0 for "Diablo not slain")
When a character creates a game it checks the binary and if 1 then i have two ideas:

1. Have Diablo suffer the Stone Curse affliction, then shatter and if possible spawn Albrechts corpse (The one on Lazarus's table will do nicely)
2. Have him do the standard bleed animation and just remove him when he is out of sight of the characters in the game.


>>Elixir - Tell me if i'm capped on them.

>probably not

A shame, but its a minor thing :)


>>Show me spell hotkeys in the UI.

>open speed book

I imagined a UI ala this:
![UI](https://user-images.githubusercontent.com/53155671/221516290-397408f8-c1c7-461c-8190-8c510a5ec68f.png)


>>Enemy type under health bar - Demon, undead, animal

>enable in settings

I don't think this setting is in the current version? I've checked - Thrice!

>>Remove grief - Black deaths, Bad shrines, broken items.

>no, enable in settings, no

How come? It was something i really liked about Diablo+

>>Level up message when my stats are capped.

>not so sure

Would be cool, but it's fine.

>>Undropable unique fix?

>Not easy/unlikely

I respect a "difficult to do" ;)

**Reply by AJenbo** [reply]
> Is it because hard to do or because bad idea?

We feel it's a bit much of a change compared to the original

**Reply by AJenbo** [reply]
> I don't think this setting is in the current version? I've checked - Thrice!

It's been there since day one

```ini
Enemy Health Bar=0
Show Monster Type=0
```

**Reply by AJenbo** [reply]
> I imagined a UI ala this:

There are topics dedicated to alternative UIs, I don't think this a la carte discussion works well for it.

**Reply by AJenbo** [reply]
> How come? It was something i really liked about Diablo+

Deviates from the original game, leave it for mods

**Reply by ghost** [reply]
> It's been there since day one
> 
> Enemy Health Bar=0
> Show Monster Type=0

I see, we're having a misunderstanding. 
![type](https://user-images.githubusercontent.com/53155671/221546458-85a97cfb-a36a-4eed-a27d-cf3757e143d5.png)

**Reply by AJenbo** [reply]
Thanks for the image but I think you are the one that needs to take a closer look. It's indicated in the health bar by changing the colored frame.

**Reply by ghost** [reply]
I would never have noticed if you hadn't told me.

**Comment by Kiiro-Yakumo** [comment]
If I had to guess, Crippling Shrines are not about difficulty but a situation like in tabletop game where the GM plays card with big middle finger and tells you he is allowed to do that because he is the GM. Black Deaths while super annoying no doubt, they pose challenge, more or less FAIR challenge. As for broken items, if you mean they remain like in Diablo II with 0 durability then well, on one hand it would be indeed good to have the ability to fix them, on the other remember that Diablo I was supposed to be difficult, we already have quite a lot of QoL enhancements available. Diablo II was MUCH easier with or without fixable items from point 0.

EDIT - Oh and let's not talk about how easy Diablo III is...

**Reply by ghost** [reply]
> As for broken items, if you mean they remain like in Diablo II with 0 durability

Yes, that's exactly what i mean. It's just annoying that they disappear. I just don't think these grief mechanics are challenging or fun - Just annoying. I certainly don't want the game to be easier, difficulty is fine as is.

1. **Broken items** - Diablo II style would work fine.
2. **Shrines** - Diablo+ style. Like, remove a few magic points instead of total mana. You can get magic points back with elixirs after all.
3. **Black deaths** - Temporary debuff or remove vitality ala. Diablo+

Kphoenix made Diablo+ and afaik he is working on this project now. Should be easy to implement.

**Reply by AJenbo** [reply]
Maybe you will like the mod that KP is currently working on using DevilutionX 1.5 as a basis

**Reply by ghost** [reply]
Possibly? DevX is great though, much appreciate the effort going into this.
